docs/data/toolpad/connecting-to-datasources/connections.md

```diff
@@ -0,0 +1,86 @@
+# Connections
+
+<p class="description">
+    <b>CONNECTION</b> instance allows us to reuse certain configuration details to create <b>queries</b> faster.
+</p>
+
+In order to create a new **CONNECTION** instance press **+** button in the **Explorer** on the left:
+
+![Add Connection](/static/toolpad/connection.png)
+
+## Connection types
+
+There are 2 datasources that **do not require** **CONNECTION** instance:
+
+1. [Function](/toolpad/connecting-to-datasources/function/) datasource:
+
+   ![Connection function](/static/toolpad/connection-function-1.png)
+
+   We can define **Secrets** (key/value) that we can later access in the **Function** datasource **query**:
+
+   ![Connection function](/static/toolpad/connection-function-2.png)
+
+1. [Fetch](/toolpad/connecting-to-datasources/fetch/) datasource:
+
+   ![Connection fetch](/static/toolpad/connection-fetch-1.png)
+
+   We can configure:
+
+   - **base url** - url that can be shared between different queries.
+   - **Headers** - pass custom headers with each request.
+   - **Authentication** type - if your API is protected choose authentication method that fits your needs:
+     - **Basic** - adds **Authorization** header and generates **base64** encoded value for a given user credentials.
+     - **Bearer token** - adds **Authorization** header generates value for provided token.
+     - **API key** - adds key/value header.
+
+There are 2 datasources that **require** **CONNECTION** instance:
+
+1. [Google Sheets](/toolpad/connecting-to-datasources/google-sheets/) datasource:
+
+   **Obtain credentials**
+
+   Before we create a **Google sheets** connection we must first configure Toolpad app by providing Google sheets **client id** and **secret**:
+
+   1. Create a new Google Cloud project - [instructions](https://cloud.google.com/resource-manager/docs/creating-managing-projects#creating_a_project).
+
+   2. Generate Oauth 2.0 credentials - [instructions](https://developers.google.com/identity/protocols/oauth2/web-server#creatingcred).
```

Comment:
I'm not sure why this whole section of the doc is indented like this. I would remove the indentations for the remaining lines as well.

```suggestion
2. [Generate Oauth 2.0 credentials](https://developers.google.com/identity/protocols/oauth2/web-server#creatingcred).
```